varnish-sys/src/vcl/backend.rs

```diff
@@ -119,11 +117,12 @@ impl<S: Serve<T>, T: Transfer> Backend<S, T> {
     /// Create a new builder, wrapping the `inner` structure (that implements `Serve`),
     /// calling the backend `name`. If the backend has a probe attached to it, set `has_probe` to
     /// true.
-    pub fn new(ctx: &mut Ctx, name: &str, be: S, has_probe: bool) -> VclResult<Self> {
+    pub fn new(ctx: &mut Ctx, type_: &str, name: &str, be: S, has_probe: bool) -> VclResult<Self> {
```

Comment:
perhaps use `ctype` here too? or give it a more descriptive name (like `backend_id`?)

Reply:
`backend_type`? it should be the same for all backend of a same category

Reply:
bikeshedding the name a bit more - `type` here seem to mean "the name of the struct that represents a class of  backends", right? Or is this a specific instance of a backend?  So perhaps `backend_info` or `backend_id` or ... ?  `type` is a bit confusing here...

Reply:
`backend_category`? :)

Reply:
I don't really like going with `id` here as it suggests (to me at least) some uniqueness.
`backend_info` works for me, I'll do that

Reply:
`category` works to

Reply:
how is this different from "backend name" ?

Reply:
in short - do we really need to have two names here?

Reply:
there's already `name` which is the name of the backend as it will appear in `varnishadm backend.list`, and usually, it's the name of the VCL object representing the object.

So you can create multiple backends from the same vmod, and you would expect them to be from the same category but to have different names

Reply:
so one is the name given by the user, i.e. a backend "variable" -- in this case `polly`:

```vcl
new polly = be.parrot("I'll repeat this on and on");
```

and the other is the backend type, i.e. `parrot`?   Although backend and object are technically not the same thing... So anyway, maybe `be_type` and `be_name` as params?

Side thought - seems like we should have called it `#[object_name]` or `#[obj_name]` instead of `#[vcl_name]`... (we could rename it and support both - not difficult)...

Reply:
`be_type`/`be_name` it is

Reply:
would it make sense to call it `backend_id` instead of `backend_name`?  Does "be_name" imply any kind of uniqueness?

Reply:
id feels more numerical? the name is expected to be unique I believe.
I can change it, no strong opinions

Reply:
name? I thought name is arbitrary and ok to dup, whereas id is more of a unique thing -- but I guess i spent far too much time in  higher level langs for an ID to frequently mean a unique string...  perhaps `backend_id` and `backend_type` are the cleanest?  Also, note that people try to spell things out for public apis (i often ignore that, but... i shouldn't :) )